cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller.go

```diff
@@ -366,7 +366,9 @@ func (c *machineController) machineSetNodeGroups() ([]*nodegroup, error) {
 		if err != nil {
 			return err
 		}
-		if ng.MaxSize()-ng.MinSize() > 0 && pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
+		if ng.scalableResource.CanScaleFromZero() {
```

Comment:
Seems like we use CanScaleFromZero everywhere we use MaxSize - MinSize, probably could make that a function.

Reply:
Yep - need to see if the notion 'Scale from zero' is going to be sticky. If, as we talked about, we only need the annotations then all of these checks can go.

Reply:
And when I say 'we only need the annotations', I mean the following:

```yaml
apiVersion: v1
items:
- apiVersion: machine.openshift.io/v1beta1
  kind: MachineSet
  metadata:
    annotations:
      machine.openshift.io/cluster-api-autoscaler-node-group-min-size: "0"
      machine.openshift.io/cluster-api-autoscaler-node-group-max-size: "6"
      machine.openshift.io/instance-cpu-capacity: "2"
      machine.openshift.io/instance-memory-capacity: 8G
      machine.openshift.io/instance-pod-capacity: "100"
```

Reply:
Are min/max already annotations or are these part of the machineset spec?

Reply:
> Are min/max already annotations or are these part of the machineset spec?

Already annotations. This is what we use in 4.1 today.